scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -284,6 +288,34 @@ object Execution {
   }
 
   /**
+    * This is a Thread used as a shutdown hook to clean up temporary files created by some Execution
+    */
+  case class TempFileCleanup(filesToCleanup: mutable.Set[String], mode: Mode) extends Thread {
+    val LOG = LoggerFactory.getLogger(this.getClass)
+
+    override def run(): Unit = {
+      val fs = mode match {
+        case localMode: CascadingLocal => FileSystem.getLocal(new Configuration)
+        case hdfsMode: HadoopMode =>  FileSystem.get(hdfsMode.jobConf)
+      }
+
+      filesToCleanup.foreach { file: String =>
+        try {
+          val path = new Path(file)
+          if (fs.exists(path)) {
+            // The "true" parameter here indicates that we should recursively delete everything under the given path
+            fs.delete(path, true)
+          }
+        } catch {
+          // If we fail in deleting a temp file, log the error but don't fail the run
+          case e: Exception => LOG.info(s"Unable to delete temp file $file")
```

Comment:
can we pass e to the log statement? that way we get the message and stacktrace? Also should this be a warn?